src/s3ql/fsck.py

```diff
@@ -128,15 +128,16 @@ def check_foreign_keys(self):
 
         for (table,) in self.conn.query("SELECT name FROM sqlite_master WHERE type='table'"):
             for row in self.conn.query('PRAGMA foreign_key_list(%s)' % table):
-                sql_objs = { 'src_table': table,
+                sql_objs = {'src_table': table,
                             'dst_table': row[2],
                             'src_col': row[3],
-                            'dst_col': row[4] }
+                            'dst_col': row[4],
+                            'dbf': self.conn.fixes['dbf']}
 
                 for (val,) in self.conn.query('SELECT %(src_table)s.%(src_col)s '
                                               'FROM %(src_table)s LEFT JOIN %(dst_table)s '
-                                              'ON %(src_table)s.%(src_col)s = %(dst_table)s.%(dst_col)s '
-                                              'WHERE %(dst_table)s.%(dst_col)s IS NULL '
+                                              'ON %(dbf)s%(src_table)s.%(src_col)s = %(dbf)s%(dst_table)s.%(dst_col)s '
```

Comment:
Thanks for working on this!

Could you explain what the `dbf` prefix (here and elsewhere) is good for? I did not see this mentioned anywhere in https://sqlite.org/forum/forumpost/031e262a89b6a9d2.

Reply:
It's just the `+` that is necessary to disable using indexes. It gets set in database.py when we detect that it is needed.

(dbf = disable bloom filter)